raijin_scripts/deploy/build_environment_module.py

```diff
@@ -63,40 +94,92 @@ def prep(config_path):
 
 
 def date(date_format="%Y%m%d") -> str:
+    """
+    Return datatime format as YYYYMMDD
+
+    :param date_format: Date format as a str
+    :return: datetime as a str
+    """
     return datetime.datetime.now().strftime(date_format)
 
 
-def run(cmd: str):
-    LOG.debug('Running command: %s', cmd)
-    return subprocess.run(cmd, shell=True, check=True, stdout=sys.stdout, stderr=sys.stderr)
+def run_command(cmd):
+    """
+    Run subprocess command and print the output on the terminal and the log file
+
+    :param cmd: Command to execute
+    :return: None
+    """
+    try:
+        LOG.debug('Running command: %s', cmd)
+        proc_output = subprocess.run(cmd, shell=True, check=True,
+                                     stdout=subprocess.PIPE,
+                                     stderr=subprocess.STDOUT,
+                                     universal_newlines=True,
+                                     encoding='utf-8',
+                                     errors='replace')
+
+        for line in proc_output.stdout.split(os.linesep):
+            LOG.debug(line)
+    except subprocess.CalledProcessError as suberror:
+        LOG.exception("Failed : %s", suberror.stdout)
 
 
 def install_conda_packages(env_file, variables):
+    """
+    Install required conda packages specified in environment yaml file
+
+    :param env_file: Environment yaml file path
+    :param variables: Configuration variables as per configuration settings in modulespec yaml file
+    :return: None
+    """
     LOG.debug('Installing conda packages from %s', env_file)
 
-    env_name = variables['module_name']
     conda_path = variables['conda_path']
     module_path = variables['module_path']
 
-    run(f"{conda_path} env create -p {module_path} -v --file {env_file}")
+    run_command(f"{conda_path} env create -p {module_path} -v --file {env_file}")
 
 
 def write_template(template_file, variables, output_file):
+    """
+    Create a new dea module file using template file
+
+    :param template_file: Module template file
+    :param variables: Configuration variables as per configuration settings in modulespec yaml file
+    :param output_file: New dea module file
+    :return: None
+    """
     LOG.debug('Filling template file %s to %s', template_file, output_file)
     LOG.debug('Ensuring parent dir %s exists', output_file.parent)
     output_file.parent.mkdir(parents=True, exist_ok=True)
+    sleep(5)  # sleep 5 seconds
```

Comment:
This shouldn't be necessary...

There are sometimes filesystem sync issues on VDI, but I'm not aware of them on `raijin`.

Not that it's a *huge* problem. But it is ugly and appears fragile.

Reply:
Removed 5 seconds wait.